examples/core/Graphormer/train.py

```diff
@@ -392,13 +393,16 @@ def collate(self, samples):
         node_feat = []
         in_degree, out_degree = [], []
         path_data = []
+        # use -1 padding since shortest_dist returns -1 for unreachable node pairs
         dist = -th.ones(
             (num_graphs, max_num_nodes, max_num_nodes), dtype=th.long
         )
 
         for i in range(num_graphs):
+            # positions outside num_nodes should be masked with non-zero values
             attn_mask[i, :, num_nodes[i] + 1 :] = 1
 
+            # +1 to distinguish the padding value and the node feature
```

Comment:
Ditto

Reply:
Node feature padding is done by `pad_sequence` below, which will pad invalid positions with zero node features.

Reply:
It might be more clear to say "to distinguish padded non-existing nodes from real nodes"?